### ISSUE TITLE
The route created while creating the application in a multi-cluster environmen failed

### ISSUE THREAD

**Issue by wenxin-01** [kubesphere/kubesphere#4396]
**Versions Used**
KubeSphere:   `v3.2.0-rc.1`

**How To Reproduce**
1.  Create app 'test-nginx'
2. Create Stateless Service  use image 'nginx'
3.  Add route role
4. Clcik on 'Create'

![image](https://user-images.githubusercontent.com/68640256/138862311-944b6b06-d563-460f-bd75-6ce4f12a75aa.png)
![image](https://user-images.githubusercontent.com/68640256/138862355-edb136a5-ee29-465a-a2af-4ad0240ca565.png)

```
apiVersion: types.kubefed.io/v1beta1
kind: FederatedIngress
metadata:
  annotations:
    kubesphere.io/creator: admin
  finalizers:
    - kubefed.io/sync-controller
  labels:
    app.kubernetes.io/name: test-nginx
    app.kubernetes.io/version: v1
  name: test-nginx-ingress-q3aqpv
  namespace: multi
spec:
  overrides:
    - clusterName: host
      clusterOverrides:
        - path: /metadata/annotations
          value: []
        - path: /spec/rules
          value:
            - clusters:
                - host
                - member
              host: test-nginx.com
              http:
                paths:
                  - backend:
                      service:
                        name: test-service
                        port:
                          number: 80
                    path: /
                    pathType: ImplementationSpecific
              protocol: http
        - path: /spec/tls
          value: []
    - clusterName: member
      clusterOverrides:
        - path: /metadata/annotations
          value: []
        - path: /spec/rules
          value:
            - clusters:
                - host
                - member
              host: test-nginx.com
              http:
                paths:
                  - backend:
                      service:
                        name: test-service
                        port:
                          number: 80
                    path: /
                    pathType: ImplementationSpecific
              protocol: http
        - path: /spec/tls
          value: []
  placement:
    clusters:
      - name: host
      - name: member
  template:
    metadata:
      labels:
        app.kubernetes.io/name: test-nginx
        app.kubernetes.io/version: v1
      namespace: multi
    spec:
      rules:
        - clusters:
            - host
            - member
          host: test-nginx.com
          http:
            paths:
              - backend:
                  service:
                    name: test-service
                    port:
                      number: 80
                path: /
                pathType: ImplementationSpecific
          protocol: http
      tls: []

```

**Expected behavior**

/kind bug
/priority high
/milestone v3.2

**Comment by iawia002** [comment]
Same reason as https://github.com/kubesphere/kubesphere/issues/4343#issuecomment-940706667

```
        - path: /metadata/annotations
          value: []
```

the structure of annotations should be map, not list.

/cc @kubesphere/sig-console

**Comment by harrisonliu5** [comment]
fixed by PR https://github.com/kubesphere/console/pull/2444

**Comment by wenxin-01** [comment]
issue fixed in `3.2.1-rc.2`
